python_ta/debug/snapshot_manager.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+import inspect
+import logging
+import os
+import re
+import sys
+import types
+from typing import Any, Iterable, Optional
+
+from python_ta.debug.snapshot import snapshot
+
+
+class SnapshotManager:
+    """
+    A class used for snapshot-based debugging to analyze memory usage at each line in the calling function.
+
+    Instance attributes:
+        memory_viz_args: The arguments to pass to the MemoryViz CLI.
+        memory_viz_version: The version of MemoryViz to use.
+        include: A collection of function names, either as strings or regular expressions, whose variables will be captured.
+        output_directory: The directory where the memory model diagrams will be saved. Defaults to the current directory.
+    """
+
+    memory_viz_args: Optional[list[str]]
+    memory_viz_version: Optional[str]
+    _snapshot_counts: int
+    include: Optional[Iterable[str | re.Pattern]]
+    output_directory: Optional[str]
+
+    def __init__(
+        self,
+        memory_viz_args: Optional[list[str]] = None,
```

Comment:
In terms of the API design, please modify use the following:

```
def __init__(
    self,
    output_directory: Optional[str] = None,
    **kwargs
)
```

The documentation should specify that all keyword arguments are passed to the `snapshot` function, and direct users to view the documentation of that function. That will make this class a bit more robust to changes in the `snapshot` API, since we won't need to change it in two places.

Similarly I don't think you need separate instance attributes for each `snapshot` argument, just one `dict` attribute.

Please do keep the warning about the the `--output` flag though, that's good.